src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -2942,9 +2942,11 @@ void ApplicationManagerImpl::UnregisterApplication(
     auto it_app = applications_.begin();
     while (applications_.end() != it_app) {
       if (app_id == (*it_app)->app_id()) {
-        connection_handler().GetDeviceID((*it_app)->mac_address(), &handle);
-        app_to_remove = *it_app;
-        applications_.erase(it_app++);
+        if (connection_handler().GetDeviceID((*it_app)->mac_address(),
```

Comment:
Are you sure that the app still shouldn't be removed if GetDeviceID fails? Also, shouldn't there be a `break` at the end of this statement? It seems we shouldn't reach this case multiple times.

On that note, couldn't the `GetDeviceID` be called outside of this loop, with `app_to_remove->mac_address()` as the argument?